### DISCUSSION TITLE
Why is MapBox used?

### DISCUSSION THREAD

**Discussion by r3df0chs** [photoview/photoview#709] [General]
Hi,
i really like photoview but i have one little problem/question.
Why is it necessary that the maps feature is based on MapBox? Other tools (photoprism, PiGallery2, etc.) works without an extra account and API to an map service?

I dont wanna create an additional footprint/account on an external service (privacy) for my photos.

Woudnt it be possible to switch the Places feature to OSM? In every other aspect photoview is awesome.

**Comment by tmanok** [comment]
Just to add my 2c, moving from Piwigo I was a bit shocked to realize a 3rd party account would be necessary to use the Places feature in Photoview. While Piwigo's map is actually made by community members unaffiliated with Piwigo, it somehow far outpaces any expectations I've ever had, from clustering images to improve loading speed when zoomed out, to an incredible number of configuration options to change sizing, position, map layers, etc on the map. I've added a few screenshots to demonstrate this. 
![Screenshot from 2024-10-26 16-16-53](https://github.com/user-attachments/assets/71151f7a-8431-4ed9-b523-1e5d47b19fa8)
![Screenshot from 2024-10-26 16-15-37](https://github.com/user-attachments/assets/85038889-2906-4b2b-aa84-c7efc8bd4049)
![Screenshot from 2024-10-26 16-15-01](https://github.com/user-attachments/assets/a853222e-2462-48f3-96c9-feb0ea0a54da)

https://github.com/Piwigo/piwigo-openstreetmap
https://github.com/Piwigo/piwigo-openstreetmap/wiki

**Comment by kkovaletp** [comment]
There is an issue #879 with a similar request.

Answering the main question from the title: I don't know) I joined the project when the initial implementation was finished.
But I agree that we need to find an alternative.

We don't have a Frontend developer, so it would be great if someone could design the solution and propose a PR
